cognite/neat/workflows/steps_registry.py

```diff
@@ -0,0 +1,136 @@
+import importlib
+import inspect
+import os
+from pathlib import Path
+import sys
+from typing import Type
+
+from pydantic import BaseModel
+from cognite.neat.exceptions import InvalidWorkFlowError
+from cognite.neat.workflows.model import WorkflowConfigItem
+import cognite.neat.workflows.steps.lib
+import logging
+
+from cognite.neat.workflows.steps.step_model import DataContract, T_Output
+from cognite.neat.workflows.steps.step_model import Step
+
+
+class StepMetadata(BaseModel):
+    name: str
+    description: str = ""
+    category: str = "default"  # defines the category the step belongs to (e.g. "data", "model", "test")
+    scope: str = "global"  # defines the scope of the step (e.g. "global", local to a specific workflow)
+    input: list[str]
+    output: list[str]
+    configuration_templates: list[WorkflowConfigItem] = []
+
+
+class StepsRegistry:
+    def __init__(self, data_store_path: str | None = None):
+        self._step_classes: list[Step] = []
+        self.user_steps_path = Path(data_store_path) / "steps"
+        self.data_store_path = data_store_path
+
+    def load_step_classes(self):
+        if self._step_classes:
+            # classes already loaded - no need to reload
+            return
+        for name, step_cls in inspect.getmembers(cognite.neat.workflows.steps.lib):
+            if inspect.isclass(step_cls):
+                self._step_classes.append([name, step_cls])
```

Comment:
```suggestion
                self._step_classes.append(step_cls)
```
Name is available with `step_cls.__name__`.